clang/lib/CConv/Constraints.cpp

```diff
@@ -35,8 +35,25 @@ Constraint::Constraint(ConstraintKind K, std::string &Rsn,
 
 // Remove the constraint from the global constraint set.
 bool Constraints::removeConstraint(Constraint *C) {
```

Comment:
Since you are now removing an edge from the `ConstraintsGraph` does this mean you will need to update any solution it already has? Are you assuming there is no current solution?

Reply:
It is the responsibility of the caller of this method to recompute the solution. If you see `CConv.cpp` we call the solver again after removing the constraint.

Reply:
Is there a way to remember that the solution may now be inconsistent? It would be nice for avoiding future confusing bugs if we could "automatically" invalidate the solution so that code didn't attempt to use it after this point.

Reply:
Ah, based on your comment below, maybe this is now a non-issue, if the constraint graph is always recreated during solving.